### DISCUSSION TITLE
Optimize recording hours

### DISCUSSION THREAD

**Discussion by jungle-boogie** [mcguirepr89/BirdNET-Pi#449] [Q&A] [answered]
Hello,

I did a quick search and didn't find anything on this, but that doesn't mean this
 hasn't been discussed before.

I'm wondering if the software is 'smart' enough to know when bird activity would
likely occur during a range of hours.

For instance, the database probably records when birds are heard, and therefore,
a range of hours could be setup to record/monitor during various hours. Either a
consecutive range, 6am to 10am or at non-consecutive hours 6am, 8am, 2pm, 4pm, etc.

This may mean it would need a full month's worth of data record to record for the
most accurate results. Of course this could possibly mean birds heard outside the
expected hours would be missed.

Thanks!

**Comment by jds3d** [comment]
What is the harm in recording all hours?  Not sure what you're trying to
optimize for here.

**Reply by jungle-boogie** [reply]
I don't see the point in recording hours and and hours of silence or no chirping birds.

**Comment by ehpersonal38** [comment]
This thread may be relevant for you - https://github.com/mcguirepr89/BirdNET-Pi/issues/175

It's not a truly native solution yet, I think Patrick has it on his todo list to implement hence why the issue is still open.

**Comment by jds3d** [comment]
It only saves the recording if it registers a bird.  Birds migrate at
night, you will hear them during spring and fall migrations.  There is
literally no benefit to turning it off unless you want to save a few
pennies of electricity.

**Comment by jungle-boogie** [comment]
The first feature says this:

   - 24/7 recording and BirdNET-Lite analysis

Are you saying it doesn’t record the entire time?

**Comment by jds3d** [comment, marked as answer]
It's listening all the time, but it only saves the mp3 file if it finds a
bird.  I mean it probably saves it temporarily so that it can be processed,
but then it will eventually overwrite/purge it.  I didn't write the
software, but I just don't get what problem you're solving by turning it
off sometimes.

**Reply by jungle-boogie** [reply]
I see so it doesn't record 24/7, per say, more likely streaming and writing temp files.

**Reply by ehpersonal38** [reply]
Good writeup from Patrick here:

The audio lifecycle (on a default installation) is:
```
RECS_DIR=$HOME/BirdSongs
EXTRACTED=$HOME/BirdSongs/Extracted
PROCESSED=$HOME/BirdSongs/Processed

         (Raw audio from mic)
                  |
                  |
                  V
       $RECS_DIR/[nested-by-date]
                  |
                  |
                  V
       `birdnet_analysis` service
       runs the audio through the
          BirdNET-Lite model
        |                   |
        |                   |
        V                   V
extracted bird        the raw audio files
sounds go to       from which the exctrations
$EXTRACTED          were made go to $PROCESSED 
                (Only 50 of the most recents are kept, 
              older files are overwritten by newer ones)
```
[source](https://github.com/mcguirepr89/BirdNET-Pi/issues/431#issuecomment-1185746590)

**Comment by cam-el** [comment]
I am going to implement something like this but for the reason that my Pi0W2 just cannot keep up with the load of 24/7 monitoring.
I am thinking first off cron jobs to to stop/start the recording service out of "bird hrs".
The birdnet_analysis service appears to stop processing files for some reason (In my IT terms it appears to crap itself) so I will be restarting the birdnet_analysis after the recording service stops/starts until I can find a way to monitor the service and make sure it is processing audio files and not just "running" with no one home.
By Stopping the recording service I am reducing the number of read/writes to the SD card increasing io for analysis and reducing the total number of files that need to be analysed. That way the system is ready for the dawn chorus!

Ultimately I hope there is a way to control this via Home Assistant with off device monitoring executing the stock scripts included with BirdNET -Pi.

**Reply by travispearl** [reply]
I'm also running a Pi02W and finding it can't keep up with processing recordings in real time. Did you find any solutions? I increased recording length from 15s to 30s, but still running behind.  

My recording/processing services have also failed, requiring me to manually delete recordings files that were causing the processing services to hang.

Birdnetpi runs great on my Pi4, but is struggling on the 02W.

**Reply by cam-el** [reply]
no not really, even staying on top of it by monitoring manually and responding by stopping starting services etc. the 02w just falls behind and becomes bogged down. I had some Australian Magpies breeding nearby and it just gets hammered from minute long bird solos.

**Reply by MueJosh** [reply]
I have no issues with my (multiple) 02w 

Yes, it's always 2-3 minutes behind but that doesn't bother me. 
Even on busy days (2-3k detections in less than a hour) it keeps its pace. 

I'm running nothing except the analyzer and saving the audio files etc. 
No BirdWeather or Apprise stuff. Plain BirdNET-Pi 

Good cooling is really important, I'm running a full metal case for passive cooling. Just a little top cooler (passive) doesn't do the trick...
